File: sdkwork-backend-sdk-typescript/src/types/video-shot-item.ts
import type { AssetMediaResourceList } from './asset-media-resource-list';
import type { ImageMediaResource } from './image-media-resource';
import type { VideoMediaResource } from './video-media-resource';

/** Video shot item */
export interface VideoShotItem {
  /** Shot unique identifier */
  id?: string;
  /** Shot UUID identifier */
  uuid?: string;
  /** Video shot prompt */
  shotPrompt?: string;
  /** Shot dialogue content */
  dialogue?: string;
  /** Duration (seconds) */
  duration?: number;
  /** Start frame image resource */
  startFrame?: ImageMediaResource;
  /** End frame image resource */
  endFrame?: ImageMediaResource;
  /** Single image mode image resource */
  image?: ImageMediaResource;
  /** Other resources */
  resourceList?: AssetMediaResourceList;
  /** Voice emotion */
  emotion?: 'NEUTRAL' | 'HAPPY' | 'EXCITED' | 'SERIOUS';
  /** Shot video generation description */
  videoDescription?: string;
  /** Generated video resource */
  video?: VideoMediaResource;
}

File: sdkwork-backend-sdk-go/types/video_shot_item.go
package types

// Video shot item
type VideoShotItem struct {
	Id string `json:"id"`
	Uuid string `json:"uuid"`
	ShotPrompt string `json:"shotPrompt"`
	Dialogue string `json:"dialogue"`
	Duration float64 `json:"duration"`
	StartFrame ImageMediaResource `json:"startFrame"`
	EndFrame ImageMediaResource `json:"endFrame"`
	Image ImageMediaResource `json:"image"`
	ResourceList AssetMediaResourceList `json:"resourceList"`
	Emotion string `json:"emotion"`
	VideoDescription string `json:"videoDescription"`
	Video VideoMediaResource `json:"video"`
}
